src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -112,7 +112,8 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
                 else if (SystemEnvironment.Instance.IsLinuxAppService())
                 {
                     var hostNameProvider = p.GetService<HostNameProvider>();
-                    return new LinuxAppServiceEventGenerator(new LinuxAppServiceFileLoggerFactory(), hostNameProvider);
+                    IOptions<FunctionsHostingConfigOptions> functionsHostingConfigOptions = p.GetService<IOptions<FunctionsHostingConfigOptions>>();
```

Comment:
We need to validate setup is already executed for `IOptions<FunctionsHostingConfigOptions>` here

Reply:
Validated that HostingConfigOptions gets loaded up before the IEventGenerator is resolved. Discussed offline. Resolving